pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -289,40 +303,17 @@ func EndpointsEqualBeyondHash(ep1, ep2 *discovery.Endpoint) bool {
 		return false
 	}
 
-	if boolPtrChanged(ep1.Conditions.Ready, ep2.Conditions.Ready) {
+	if !apiequality.Semantic.DeepEqual(ep1.Conditions, ep2.Conditions) {
```

Comment:
don't we have the same problem of triggering updates with upgrades/enabling feature gate here ? 

https://github.com/kubernetes/kubernetes/blob/6acc62da7560c96f52617e744ee8be66913c72b2/pkg/controller/endpointslice/utils.go#L64-L67


https://github.com/kubernetes/kubernetes/blob/6acc62da7560c96f52617e744ee8be66913c72b2/pkg/controller/endpointslice/reconciler.go#L368-L377

https://github.com/kubernetes/kubernetes/blob/6acc62da7560c96f52617e744ee8be66913c72b2/pkg/controller/endpointslice/reconciler.go#L178-L182

Reply:
Good point. EndpointSliceTerminatingCondition is enabled by default since 1.22 so it should be fine for 1.23-1.24 upgrade and I suppose we don't want to backport such change. For users that toggle the feature gate intentionally, I think they are supposed to see that the fields are updated according to their configuration?

Reply:
I was considering that there are some cluster that don't enable all betas too ...